### PR TITLE
Encode passwords with asterisks

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
So that they are not outputted when --log trace option is provided
Fixes http://teampulse.telerik.com/view#item/289692

### CODE REVIEW

File: logger.ts [hunk 1]
```diff
@@ -109,5 +113,39 @@ export class Logger implements ILogger {
 
 		return printMsgFuture;
 	}
+
+	private getPasswordEncodedArguments(args: string[]): string[] {
+		return _.map(args, a => {
+			var passwordRegex = /[Pp]assword=(.*?)(['&,]|$)|\"[Pp]assword\":\"(.*?)\"/,
```
Comment (teobugslayer) [inline logger.ts]: The regexp creation should happen only once. Consider moving it outside the map lambda.

File: logger.ts [hunk 2]
```diff
@@ -109,5 +115,40 @@ export class Logger implements ILogger {
...
+					this.encodeBody = a.indexOf(path) > -1;
+					break;
+				}
+			}
+
+			return a;
+		});
+	}
+
+	private getEncodedPassword(password: string, originalString: string) {
```
Comment (teobugslayer) [inline logger.ts]: The name of this method does not reflect its function. It does not get the password, it replaces it in the string. Perhaps something similar to ```hidePassword```. Note, the password is not encoded, it is removed.

File: logger.ts [hunk 3]
```diff
@@ -109,5 +115,40 @@ export class Logger implements ILogger {
 
 		return printMsgFuture;
 	}
+
+	private getPasswordEncodedArguments(args: string[]): string[] {
+		return _.map(args, a => {
```
Comment (rosen-vladimirov) [inline logger.ts]: can we change `a` to `arg` or `argument`

File: logger.ts [hunk 4]
```diff
@@ -109,5 +115,40 @@ export class Logger implements ILogger {
...
+			}
+
+			if (this.encodeBody) {
+				var bodyMatch = this.requestBodyRegex.exec(a);
+				if (bodyMatch) {
+					return this.getEncodedPassword(bodyMatch[1], a);
+				}
+			}
+
+			for (var i = 0; i < this.encodeRequestPaths.length; ++i) {
```
Comment (rosen-vladimirov) [inline logger.ts]: why not using _.each

**Comment by teobugslayer** [comment]
:+1:

**Pull request merged**